src/components/playground/PlaygroundComponent/PlaygroundComponent.jsx

```diff
@@ -31,6 +33,28 @@ export const PlaygroundComponent = () => {
   const [selected, setSelected] = useState("Select API Call - Version 3");
   const location = useLocation();
   const history = createBrowserHistory();
+  
+  // If the user switches to a different tab, it will trigger the visibility state.
+  const documentVisibility = () => {
+    // If the visibility state is hidden, we will close the API.
+    if (document.visibilityState === "hidden") {
+      current_api.connection.close();
+      ticksSubject.complete();
+      document.removeEventListener("visibilitychange", documentVisibility, false);
+      setScrollDirection("down");
+      setVisibilityListener(true);
+    }
+    // When we switch back to the main window, initiate a new API call.
+    setCurrentAPI(api);
+  }
+
+  // Here we initialize the visibility listener.
+  useEffect(() => {
+    if (visibility_listener) {
+      setVisibilityListener(false);
+      document.addEventListener("visibilitychange", documentVisibility, false);
+    }
+  }, [visibility_listener])
```

Comment:
Should we also delete event listener after unmounting component?

Reply:
This is a very good point. Yes indeed, we should. I will to add something in regards to that. 👍